pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,447 @@
+                urljoin(repo['_href'], 'actions/sync/'),
+                {'override_config': {}}
+            ))
+            cls.sync_reports[-1].raise_for_status()
+        cls.task_bodies = tuple(chain.from_iterable(
+            utils.poll_spawned_tasks(cls.cfg, report.json())
+            for report in cls.sync_reports
+        ))
+        for repo in repos:
+            cls.resources.add(repo['_href'])
+
+    def test_start_sync_code(self):
+        """Assert the call to sync a repository returns an HTTP 202."""
+        for sync_repo in self.sync_reports:
+            with self.subTest(sync_repo=sync_repo):
+                self.assertEqual(sync_repo.status_code, 202)
+
+    def test_task_error(self):
+        """Assert each task's "error" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['error'])
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['traceback'])
+
+    def test_error_details(self):
+        """Assert that some task's progress report contains error details."""
+        for i, task_body in enumerate(self.task_bodies):
+            error_details_num = 0
+            for action in task_body['progress_report']['ostree_web_importer']:
+                if action['error_details'] != []:
+                    error_details_num += 1
+            self.assertNotEqual(error_details_num, 0)
+
+    def test_number_tasks(self):
+        """Assert that two task were spawned."""
+        self.assertEqual(len(self.task_bodies), 2)
+
+
+class PublishTestCase(_BaseTestCase):
+    """Create two repositories, one with feed.
+
+    Sync, copy first repository to the second, update branch on first and
+    publish it. Download and compare units.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an OSTree repository with a valid feed and sync it.
+
+        Create two repos, one with feed. Sync one, copy to another, publish,
+        check that units match.
+        """
+        super(PublishTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+
+        steps = {
+            'update',
+            'sync',
+            'copy',
+            'distribute',
+            'publish',
+            'search units',
+        }
+        cls.responses = {key: [] for key in steps}
+        cls.bodies = {}
+        cls.task_bodies = {key: [] for key in {'sync', 'copy'}}
+
+        bodies = tuple(_gen_ostree_repo_body() for _ in range(2))
+        bodies[0]['importer_config']['feed'] = _VALID_FEED
+        bodies[0]['importer_config']['branches'] = [_VALID_BRANCHES[0]]
+        repos = tuple(client.post(REPOSITORY_PATH, body) for body in bodies)
+        client.response_handler = api.echo_handler
```

Comment:
Why use the `echo_handler` here? If something goes wrong, we want to know as soon as possible. How about using the `safe_handler` instead?